main.go

```diff
@@ -12,6 +13,30 @@ import (
 	"golang.org/x/oauth2"
 )
 
+func setupCloseHandler(ctx context.Context, apprv *approvalEnvironment, client *github.Client) {
+	killSignal := make(chan os.Signal, 1)
+	signal.Notify(killSignal, os.Interrupt)
+	go func(ctx context.Context, apprv *approvalEnvironment, client *github.Client) {
+		<-killSignal
+		newState := "closed"
+		closeComment := "Workflow cancelled, closing issue."
+		fmt.Println(closeComment)
+		_, _, err := client.Issues.Edit(ctx, apprv.repoOwner, apprv.repo, apprv.approvalIssueNumber, &github.IssueRequest{State: &newState})
+		if err != nil {
+			fmt.Printf("error closing issue: %v\n", err)
+			os.Exit(1)
+		}
+		_, _, err = client.Issues.CreateComment(ctx, apprv.repoOwner, apprv.repo, apprv.approvalIssueNumber, &github.IssueComment{
+			Body: &closeComment,
+		})
+		if err != nil {
+			fmt.Printf("error commenting on issue: %v\n", err)
+			os.Exit(1)
+		}
+		os.Exit(1)
```

Comment:
We're still exiting the process from the non-main goroutine. I think this goroutine should just return back to the main goroutine which should handle process exiting and status codes.

I think this is just about there, but my recommendation here would be to not call `os.Exit` from this goroutine and instead use channels to communicate back to the main goroutine. And likely this would mean that we should refactor the `commentLoop` section of main into it's own goroutine so that the main goroutine is just doing a `select` on two different channels for completion or cancellation.

Reply:
I've pushed a commit that should meet your requirements;

 - All `os.Exit()` calls are now made on the main goroutine
 - The `commentLoop` is now wrapped in a goroutine that passes an exit code through a channel, handled by a select statement in `main()`
 - If the kill signal is caught, it will run the `interruptHandler` function to clean up the issue, always exiting with `os.Exit(1)`

I'm a relative novice to Golang, so feel free to critique anything you think might cause issues with scalability/best practice. I've actually learned a lot about concurrency in Golang during this PR!